rollingpin/elasticsearch.py

```diff
@@ -95,8 +137,23 @@ def deploy_end_doc(self):
         }
 
     @inlineCallbacks
+    def on_build_sync(self, sync_info):
+        yield self.update_doc(self.deploy_annotation_id,
+                              self.build_sync_doc(sync_info))
+
+    @inlineCallbacks
     def on_deploy_start(self):
-        yield self.index_doc(self.deploy_start_doc())
+
+        # Store the initial document's ID so we can update it with additional
+        # metadata later in the deploy process.
+        def read_response(response):
+            def store_deploy_annotation_id(body):
+                self.deploy_annotation_id = json.loads(body).get('_id', '')
+            d = readBody(response)
+            d.addCallback(store_deploy_annotation_id)
+            return d
+
+        yield self.index_doc(self.deploy_start_doc(), cb=read_response)
```

Comment:
Wow, yeah.  I could have sworn I tried this but maybe I had my `yield`s and my `return`s mixed up.  It seems to work now and is much cleaner.